### PR TITLE
chore(ci): fix package publish next COMPASS-6860

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch addresses CI network issues that are haunting us every time we publish packages by splitting publish command to two steps: first updates the version number, another tries to publish with retries. This works because we now use `from-package` mode for publishing that checks what versions were already published, so with a few retries, we can finish publishing everything

This task doesn't run in CI, so I did a [manual patch](https://spruce.mongodb.com/task/10gen_compass_main_ubuntu_publish_publish_packages_next_patch_b53158d1a4cfaa42b802a8e10059a81778b5ea8b_64f9ba642fbabef3600b70b8_23_09_07_11_56_41/logs?execution=0) and also tried it locally

### CODE REVIEW

File: packages/compass-app-stores/package.json [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
     "email": "compass@mongodb.com"
   },
-  "private": true,
   "bugs": {
     "url": "https://jira.mongodb.org/projects/COMPASS/issues",
     "email": "compass@mongodb.com"
   },
   "homepage": "https://github.com/mongodb-js/compass",
-  "version": "0.5.2",
+  "version": "7.0.0",
```
Comment (gribnoysup) [inline packages/compass-app-stores/package.json:14]: This was incorrectly versioned and marked as private. Fixed, version is just next major from what was last published

File: packages/compass-field-store/package.json [hunk 2]
```diff
@@ -6,13 +6,12 @@
...
     "email": "compass@mongodb.com"
   },
-  "private": true,
   "bugs": {
     "url": "https://jira.mongodb.org/projects/COMPASS/issues",
     "email": "compass@mongodb.com"
   },
   "homepage": "https://github.com/mongodb-js/compass",
-  "version": "0.3.1",
+  "version": "9.0.0",
```
Comment (gribnoysup) [inline packages/compass-field-store/package.json:14]: Same as above

File: packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.ts [hunk 3]
```diff
@@ -239,7 +239,10 @@ export default class PipelineParser {
 }
 
 function getLineOnlySourceLocation(line: number) {
-  return { start: { line, column: 0 }, end: { line, column: 0 } };
+  return {
+    start: { line, column: 0 },
+    end: { line, column: 0 },
+  } as t.SourceLocation;
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.ts:245]: This started failing when compiling compass-aggregations plugins, probably a version change somewhere that we merged recently that wasn't directly touching the package and so went unnoticed

**Review by lerouxb — APPROVED** [review]

File: packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.ts (hunk 3 above)
Comment (lerouxb) [reply packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.ts:245]: Saw the same change in the bson 6 mongosh 2 PR.

Comment (gribnoysup) [reply packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.ts:245]: Hmmm, true, probably I was jumping between branches and did `npm install` when doing changes to this bson 6 branch that affected this when I was testing it locally

**Pull request merged**